Tools/autotest/arducopter.py

```diff
@@ -672,248 +687,263 @@ def fly_throttle_failsafe(self, side=60, timeout=360):
+            self.delay_sim_time(5)
+            self.set_parameter('SIM_BATT_VOLTAGE', 11.4)
+            self.mavproxy.expect("Battery 1 is low")
+            self.delay_sim_time(5)
+            self.wait_mode("LAND")
+            self.wait_land_and_disarm()
+            self.set_parameter('SIM_BATT_VOLTAGE', 12.5)
+            self.reboot_sitl_mavproxy()
+            self.end_subtest("Completed battery failsafe with FS_OPTIONS set to continue landing")
+
+            # Trigger a critical battery condition, which triggers a land mode failsafe. Trigger an RC failure. Verify the RC failsafe is prevented from stopping the low battery landing.
+            self.start_subtest("Battery failsafe critical landing")
+            self.takeoffAndMoveAway(100,50)
+            self.set_parameter('FS_OPTIONS', 0)
+            self.set_parameter('BATT_FS_LOW_ACT', 1)
+            self.set_parameter('BATT_FS_CRT_ACT', 1)
+            self.set_parameter('FS_THR_ENABLE', 1)
+            self.delay_sim_time(5)
+            self.set_parameter('SIM_BATT_VOLTAGE', 10.0)
+            self.mavproxy.expect("Battery 1 is critical")
+            self.wait_mode("LAND")
+            self.delay_sim_time(10)
+            self.set_parameter("SIM_RC_FAIL", 1)
+            self.delay_sim_time(10)
+            self.wait_mode("LAND")
+            self.wait_land_and_disarm()
+            self.set_parameter('SIM_BATT_VOLTAGE', 12.5)
+            self.set_parameter("SIM_RC_FAIL", 0)
+            self.reboot_sitl_mavproxy()
+            self.end_subtest("Completed battery failsafe critical landing")
+
+            # Trigger low battery condition with failsafe set to terminate. Copter will disarm and crash.
+            self.start_subtest("Battery failsafe terminate")
+            self.takeoffAndMoveAway()
+            self.set_parameter('BATT_FS_LOW_ACT', 5)
+            self.delay_sim_time(10)
+            self.set_parameter('SIM_BATT_VOLTAGE', 11.4)
+            self.mavproxy.expect("Battery 1 is low")
+            self.wait_disarmed()
+            self.end_subtest("Completed terminate failsafe test")
 
-        # Trigger low battery condition with failsafe set to terminate. Copter will disarm and crash.
-        self.start_subtest("Battery failsafe terminate")
-        self.takeoffAndMoveAway()
-        self.set_parameter('BATT_FS_LOW_ACT', 5)
-        self.delay_sim_time(10)
-        self.set_parameter('SIM_BATT_VOLTAGE', 11.4)
-        self.mavproxy.expect("Battery 1 is low")
-        self.wait_disarmed()
-        self.end_subtest("Completed terminate failsafe test")
+            reset_failsafe()
 
-        self.progress("All Battery failsafe tests complete")
-        self.set_parameter('BATT_LOW_VOLT', 0)
-        self.set_parameter('BATT_CRT_VOLT', 0)
-        self.set_parameter('BATT_FS_LOW_ACT', 0)
-        self.set_parameter('BATT_FS_CRT_ACT', 0)
-        self.set_parameter('FS_OPTIONS', 0)
-        self.reboot_sitl()
+        except Exception as ex:
```

Comment:
Why not dowe hat we usually do - catch the exception, stash it away, do the reset then raise the exception again if it is not None?

Reply:
done